app/src/main/java/com/lightricks/feedexercise/network/UrlProvider.kt

```diff
@@ -9,13 +10,15 @@ interface UrlProvider {
 /**
  * Base url provider for Retrofit
  */
+@Singleton
```

Comment:
Since this implementation doesn't hold any state then there's no reason of having it as singleton. It is even "wasteful" since it use only a small scope/context. So we should not prevent the GC to collect it when no more in use.